src/PsychicStreamResponse.cpp

```diff
@@ -0,0 +1,94 @@
+#include "PsychicStreamResponse.h"
+#include "PsychicResponse.h"
+#include "PsychicRequest.h"
+
+PsychicStreamResponse::PsychicStreamResponse(PsychicRequest *request, const String& contentType)
+ : PsychicResponse(request), _buffer(NULL) {
+
+  setContentType(contentType.c_str());
+  addHeader("Content-Disposition", "inline");
+}
+
+ 
+PsychicStreamResponse::PsychicStreamResponse(PsychicRequest *request, const String& contentType, const String& name)
+  : PsychicResponse(request), _buffer(NULL) {
+
+  setContentType(contentType.c_str());
+
+  char buf[26+name.length()];
+  snprintf(buf, sizeof (buf), "attachment; filename=\"%s\"", name);
+  addHeader("Content-Disposition", buf);
+}
+
+ 
+PsychicStreamResponse::~PsychicStreamResponse()
+{
+  endSend();
+}
+
+
+esp_err_t PsychicStreamResponse::beginSend()
+{
+  if(_buffer)
+    return ESP_OK;
+
+  //Buffer to hold ChunkPrinter and stream buffer. Using placement new will keep us at a single allocation.
```

Comment:
that's clever!

Reply:
Cheers, seeing as this response type is split (begin/end) placement new suits this situation well. 
The response stream could be derived directly from the `ChunkPrinter`, however that would require the memory to be allocated during the constructor; and short lived allocations help prevent fragmentation on our constrained hardware target.

This also follows the trend of the other responses by allocating buffers once `send()` is called.